Feature/guilds styling 658 part 2

0xvangrim opened the pull request:
The second batch of styling changes includes:

- Searchbar and filter row changes on the main page
- Styling changes of the proposal editor page (there are some styling changes that needs to be done that I have left since I have some questions about the functionality)
- Changed the vote proposal box to "For" and "Against"
- Added a pre-commit script from husky
- The `useContract` had a typo for reaching the actual `abi`.

![image](https://user-images.githubusercontent.com/59825171/155800115-331bd928-291e-4748-846a-2b083a734777.png)
![image](https://user-images.githubusercontent.com/59825171/155800159-e438d57b-b475-4f7a-9a38-cfb292007cc1.png)
![image](https://user-images.githubusercontent.com/59825171/155800194-91b85f6d-6401-4384-8232-6a5d29473e6a.png)

rossneilson committed:
Merge pull request #652 from DXgovernance/develop

Develop

Commit:
continued to style with our new theme

Commit:
Merge branches 'feature/guilds-styling-658-part-2' and 'develop' of https://github.com/DXgovernance/dxvote into feature/guilds-styling-658-part-2

Commit:
createProposalpage changes

Commit:
formatted and husky

Commit:
changed cast your vote and editor page

Commit:
format

Commit:
added format and test to pre-commit

Commit:
removed console

Commit:
styled filter row and search bar

Commit:
Merge remote-tracking branch 'origin' into feature/guilds-styling-658-part-2

Commit:
fix conflicts

Commit:
change file dir

hhamud commented:
Issue here with the repetition in the vote component.
![image](https://user-images.githubusercontent.com/53880692/155852471-9f3d2ab8-fd1a-41a1-9a24-14d3cadcd4a5.png)
link is here: https://bafybeihhbxqcjmwmyf7yl72kjfs7ouuoeavwdpide243gsfqwcmgben4ri.ipfs.dweb.link/#/guilds/rinkeby/0x9cdc16b5f95229b856cba5f38095fd8e00f8edef/proposal/0x748ef1a348c2fa383a753fd5c3da54fe5d8fd4f87f7ff0c15b4f5b873ff65085

mprasanjith commented:
> Issue here with the repetition in the vote component. ![image](https://user-images.githubusercontent.com/53880692/155852471-9f3d2ab8-fd1a-41a1-9a24-14d3cadcd4a5.png) link is here: https://bafybeihhbxqcjmwmyf7yl72kjfs7ouuoeavwdpide243gsfqwcmgben4ri.ipfs.dweb.link/#/guilds/rinkeby/0x9cdc16b5f95229b856cba5f38095fd8e00f8edef/proposal/0x748ef1a348c2fa383a753fd5c3da54fe5d8fd4f87f7ff0c15b4f5b873ff65085

@hhamud I'm working on some updates to the voting components, so I think this could be addressed in that PR.

hhamud commented on src/components/Guilds/CreateProposalPage/ActionsBuilder/ActionsBuilder.tsx:
```diff
@@ -48,11 +48,21 @@ export const ActionsBuilder: React.FC<ActionsBuilderProps> = ({
       header={
         <Flex direction="row" justifyContent="space-between">
-          <CardHeader>Actions</CardHeader>
+          <CardHeader color="#fff">Actions</CardHeader>
```
Shouldn't colors be linked to themes?. best to avoid setting this inline as well.

hhamud commented on src/components/Guilds/CreateProposalPage/SidebarInfoCard/SidebarInfoCard.tsx:
```diff
@@ -43,11 +43,13 @@ export const SidebarInfoCard = () => {
         <Row>
           <Label>Consensus System</Label>
-          <Label>Guild</Label>
+          <Label color={({ theme }) => theme.colors.proposalText.lightGrey}>
```
same thing about setting this inline.

hhamud commented on src/components/Guilds/CreateProposalPage/SidebarInfoCard/SidebarInfoCard.tsx:
```diff
@@ -43,11 +43,13 @@ export const SidebarInfoCard = () => {
         <Row>
           <Label>Proposal Duration</Label>
-          <Label>
+          <Label color={({ theme }) => theme.colors.proposalText.lightGrey}>
```
same thing about setting this inline.

hhamud commented on src/components/Guilds/CreateProposalPage/SidebarInfoCard/SidebarInfoCard.tsx:
```diff
@@ -57,7 +59,7 @@ export const SidebarInfoCard = () => {
         <Row>
           <Label>Quorum</Label>
-          <Label>
+          <Label color={({ theme }) => theme.colors.proposalText.lightGrey}>
```
same thing about setting this inline. its much easier to create a wrapper component and group these as they are all quite similiar. DRY

hhamud commented on src/components/Guilds/Filter/index.tsx:
```diff
@@ -88,8 +86,51 @@ export const Filter = () => {
+            <StyledIconButton iconRight variant="secondary">
+              Currency
+              <FaChevronDown style={{ marginLeft: 15 }} />
+            </StyledIconButton>
```
same thing about setting this inline. There is also an SVG wrapper component here that you can use [here](https://github.com/DXgovernance/dxvote/blob/master/src/components/Guilds/common/SVG/index.tsx).

hhamud commented on src/components/Guilds/ProposalPage/ProposalVoteCard/ProposalVotes.tsx:
```diff
@@ -42,23 +54,41 @@ export const ProposalVotes: React.FC<ProposalVotesProps> = ({
-  ): number =>
-    showToken ? formatUnits(args[action][action][0]) : args[action][action][1];
+  ): number => {
+    if (args[action] === null || args[action] === undefined) {
+      return 0;
+    }
```
good catch

hhamud commented on src/components/Guilds/ProposalPage/ProposalVoteCard/ProposalVotes.tsx:
```diff
@@ -42,23 +54,41 @@ export const ProposalVotes: React.FC<ProposalVotesProps> = ({
+            <>
+              <VotesRow key={i} type="0">
+                <StyledContainer>
+                  <ColoredBullet color="#295FF4">{'\u2022'}</ColoredBullet>
```
same thing about setting colours inline. should use themeing

hhamud commented on src/components/Guilds/ProposalPage/ProposalVoteCard/ProposalVotes.tsx:
```diff
@@ -42,23 +54,41 @@ export const ProposalVotes: React.FC<ProposalVotesProps> = ({
+              </VotesRow>
+              <VotesRow type="1">
+                <StyledContainer>
+                  <ColoredBullet color="#E75C5C">{'\u2022'}</ColoredBullet>
```
same thing about setting colours inline. should use themeing

hhamud reviewed (commented):
Overall code is good, it just needs a few changes. @mprasanjith has said that he will handle the vote component's repetition bug.

mprasanjith commented:
Btw a little tip if you want to access the theme config from JS, styled-components has this nice hook which exposes it.

```
import { useTheme } from 'styled-components';

const theme = useTheme();
```

0xvangrim commented:
> Overall code is good, it just needs a few changes. @mprasanjith has said that he will handle the vote component's repetition bug.

Thanks! I will take a look at your feedback and come back if I have additional questions.

Commit:
moved inline styling to components, changed voting naming

Commit:
redid the filter buttons to actually use our dropdowns

0xvangrim commented:
@hhamud @mprasanjith 

I've addressed your feedback and the last commits have fixed these issues:

- Moved inline + color-styling to wrapped components.
- Changed naming of votes to use "for" and "against" instead of the indices.
- Bigger refactoring of our filters to use our `FilterMenu` (I realized that I only styled it as buttons previously). Most of it is only additions and renamings though FYI.

Let me know if there's anything else you find 😄

hhamud reviewed (approved):
LGTM!
